toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexStartupActivity.kt

```diff
@@ -5,24 +5,50 @@
 
 package org.rust.toml.crates.local
 
+import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.startup.StartupActivity
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtilCore
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.openapi.vfs.newvfs.BulkFileListener
+import com.intellij.openapi.vfs.newvfs.RefreshQueue
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
 import com.intellij.openapiext.isUnitTestMode
 import org.rust.cargo.project.model.cargoProjects
 import org.rust.ide.experiments.RsExperiments
 import org.rust.openapiext.isFeatureEnabled
+import java.io.File
+import java.nio.file.Paths
 
 /**
  * [CratesLocalIndexService] initializer.
- * Only tries to get service instance, without any updates to its crates index.
+ * Tries to get a service instance and binds crates local index update cargo registry index repository changes.
  */
 class CratesLocalIndexStartupActivity : StartupActivity.Background {
     override fun runActivity(project: Project) {
         if (isUnitTestMode) return
         if (!project.cargoProjects.hasAtLeastOneValidProject) return
 
         if (isFeatureEnabled(RsExperiments.CRATES_LOCAL_INDEX)) {
-            CratesLocalIndexService.getInstance()
+            val cratesService = CratesLocalIndexService.getInstance()
+
+            val userCargoRegistryIndexLocation = Paths.get(System.getProperty("user.home"), CratesLocalIndexServiceImpl.CARGO_REGISTRY_INDEX_LOCATION).toString()
+            LocalFileSystem.getInstance().addRootToWatch(userCargoRegistryIndexLocation, true)
+
+            val root = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(File(userCargoRegistryIndexLocation)) ?: return
+            VfsUtilCore.processFilesRecursively(root) { true }
+            RefreshQueue.getInstance().refresh(true, true, null, root)
```

Comment:
Looks like we always can work with `Path` and don't convert them into `String` and `File`

Reply:
I didn't find a way to pass `Path` in every case, so I made a `Path` and used it's `Srtring` representation everywhere